### DISCUSSION TITLE
how to set router.replace not reset scrollbar postion?

### DISCUSSION THREAD

**Discussion by yunxifd** [vuejs/router#1813] [Help and Questions] [answered]
I want to keep some data in querystring. 
 The effect is similar as [ahooks useUrlState](https://ahooks.js.org/hooks/use-url-state?count=5#default-usage)。so i find an alternative  for vue [vueuse useRouteQuery](https://vueuse.org/router/useRouteQuery/)

vueuse useRouteQuery example
```js 
<template>
  <div>Home</div>
  <div style="height: 500px"></div>
  <input v-model="name" />
</template>
<script setup lang="ts">
import { useRouteQuery } from '@vueuse/router';
var name = useRouteQuery('name');
</script>
```
vue-router replace example
```js
<template>
  <div>Home</div>
  <div style="height: 500px"></div>
  <input v-model="name" />
  <RouterLink to="/about">go to about page</RouterLink>
</template>
<script setup lang="ts">
import { useRouter } from 'vue-router';
import { ref, watch } from 'vue';
var router = useRouter();
var name = ref('');
watch(
  () => name.value,
  () => {
    router.replace({
      query: {
        name: name.value,
      },
    });
  },
);
</script>

```
but when input , scrollbar postion reset to top

## expect
when input, keep scrollbar position.

**Comment by yunxifd** [comment]
when i  use [vueuse useUrlSearchParams](https://vueuse.org/core/useUrlSearchParams) hook（which use history.replace ），the url change and scrollbar position keep not change . but when i go to other page and goback the url not i expected

from reply: https://github.com/vuejs/router/issues/1678#issuecomment-1409962918
we should not use history.replay and vue-router together

**Comment by posva** [comment, marked as answer]
If you are seeing this behavior, it means you have a `scrollBehavior()` that scrolls up by default. You should probably tweak the behavior to not return the scroll up object in th scenario you want it to stay where it is. You could also add a custom query param or hash like `#noscroll` to completely avoid scroll behavior: `if (to.hash === '#noscroll') return` (within scrollBehavior)
